service/matching/task_reader.go

```diff
@@ -111,18 +115,39 @@ func (tr *taskReader) Signal() {
 	}
 }
 
+func (tr *taskReader) updateBacklogAge(task *internalTask) {
+	if task.event.Data.CreateTime == nil {
+		return // should not happen but for safety
+	}
+	ts := timestamp.TimeValue(task.event.Data.CreateTime).UnixNano()
+	// update the backlogAge value
+	tr.backlogHeadCreateTime.Store(ts)
+}
+
+func (tr *taskReader) getBacklogHeadAge() time.Duration {
+	if tr.backlogHeadCreateTime.Load() == -1 {
+		return time.Duration(0)
+	}
+	return time.Since(time.Unix(0, tr.backlogHeadCreateTime.Load()))
+}
+
 func (tr *taskReader) dispatchBufferedTasks(ctx context.Context) error {
 	ctx = tr.backlogMgr.contextInfoProvider(ctx)
 
 dispatchLoop:
 	for ctx.Err() == nil {
+		if len(tr.taskBuffer) == 0 {
+			// reset the atomic since we have no tasks from the backlog
+			tr.backlogHeadCreateTime.Store(-1)
+		}
 		select {
 		case taskInfo, ok := <-tr.taskBuffer:
 			if !ok { // Task queue getTasks pump is shutdown
 				break dispatchLoop
 			}
 			task := newInternalTaskFromBacklog(taskInfo, tr.backlogMgr.completeTask)
 			for ctx.Err() == nil {
+				tr.updateBacklogAge(task)
```

Comment:
are `backlogAge` and `backlogHeadCreateTime` the same thing? If so, the naming should be consistent. If two terms are used for the same value, and the reasoning for that is not clearly explained, future code readers will be slowed down because each of them will waste time trying to ask themselves "are these two things the same? if they're the same, why are they named differently? I need to figure out the difference"

Reply:
hmm, I agree yeah. To resolve this, I have made changes to have the atomic be `backlogHeadAge` which is consistent with the function names and the call made from physical_task_queue_manager.go

Reply:
I think age is duration and backlogHeadCreateTime stores a timestamp. so the current names make sense. we can add a comment to clarify though.

Reply:
I shall be marking this as resolved. Spoke to Carly async and the above reasoning made sense.